lib/perl/Genome/Model/Tools/EpitopePrediction/NewPipeline.pm

```diff
@@ -0,0 +1,192 @@
+        %{$self->dag->constant_values},
+    };
+}
+
+sub dag {
+    my $self = shift;
+
+    my $dag = Genome::WorkflowBuilder::DAG->create(
+        name => 'Epitope Binding Predicition',
+    );
+
+    $self->add_reports_to_workflow($dag);
+
+    return $dag;
+}
+
+sub add_reports_to_workflow {
+    my ($self, $dag) = @_;
+
+    for my $variant_type (qw(snvs indels)) {
+        my $input_vcf = $self->input_vcf($variant_type);
+        next unless -s $input_vcf;
+
+        my %report_params = (
+            input_vcf => $input_vcf,
+            variant_type => $variant_type,
+            plan_file => $self->plan_file($variant_type),
+            translations_file => $self->translations_file,
+        );
+        my $report = Genome::VariantReporting::Command::CreateReport->create(%report_params);
+        my $report_dag = $report->dag;
+        $dag->add_operation($report_dag);
+        $dag->connect_input(
+            input_property => 'process_id',
+            destination => $report_dag,
+            destination_property => 'process_id',
+        );
+
+        my $output_directory = File::Spec->join($self->output_directory, $variant_type);
+        Genome::Sys::make_path($output_directory);
+        my %netmhc_params = (
+            output_directory => $output_directory,
+            epitope_length => $self->epitope_length,
+            netmhc_version => $self->netmhc_version,
+            output_filter => $self->output_filter,
+            sample_name =>  $self->build->subject_name,
+            alleles => $self->alleles,
+            variant_type => $variant_type,
+        );
+        my $netmhc = Genome::Model::Tools::EpitopePrediction::NetmhcPipeline->create(%netmhc_params);
+        my $netmhc_dag = $netmhc->_construct_workflow;
+        my %netmhc_inputs = %{$netmhc->_get_workflow_inputs};
+        delete $netmhc_inputs{input_fasta_file};
+        $netmhc_dag->declare_constant(%netmhc_inputs);
+        $dag->add_operation($netmhc_dag);
+
+        for my $report_output_name ($report_dag->output_properties) {
+            if ($report_output_name =~ m/report_path \((.*)\)/) {
+                my $report_name = $1;
+                $dag->create_link(
+                    source => $report_dag,
+                    source_property => $report_output_name,
+                    destination => $netmhc_dag,
+                    destination_property => 'input_fasta_file',
+                );
+                $dag->connect_output(
+                    output_property => sprintf('%s_result (%s)', $variant_type, $report_name),
+                    source => $netmhc_dag,
+                    source_property => 'output_file',
+                );
+            }
+        }
+    }
+
+    return;
+}
+
+sub translations_file {
+    my $self = shift;
+    my $translations_file = Genome::Sys->create_temp_file_path;
+    my %translations;
+    $translations{reference_fasta} = $self->build->tumor_build->reference_sequence_build->full_consensus_path("fa");
```

Comment:
`Genome::Model::Build::SomaticVariation` has a `reference_sequence_build` method defined.  (It goes through the normal, but the build will be `Unstartable` if the two don't match.)